### PR TITLE
Add type annotation printer (ver2)

### PR DESCRIPTION
**Pull request by hide-kawabata**
Type annotation printer. When running with --debug-option=aaa.html, it prints clauses with type annotations at the bottom of the section "Ast Analysis [type-analysis]" in aaa.html (below "-- Result (2) --"). Only two files (Type.{cpp,h}) are modified.

### CODE REVIEW

File: src/ast/analysis/Type.cpp [hunk 1]
```diff
@@ -522,6 +529,249 @@ void TypeAnalysis::run(const TranslationUnit& translationUnit) {
...
+                os << ",";
+            }
+        }
+        os << ")";
+    }
+}
+
+void TypeAnnotationPrinter::print_(
+        type_identity<UserDefinedFunctor>, [[maybe_unused]] const UserDefinedFunctor& fun) {
+    // TODO
```
Comment (b-scholz) [inline src/ast/analysis/Type.cpp:677]: Can we add the printing for User-Defined Functor (perhaps we add the signature??)?

File: src/ast/analysis/Type.cpp [hunk 2]
```diff
@@ -522,6 +529,249 @@ void TypeAnalysis::run(const TranslationUnit& translationUnit) {
...
+}
+
+void TypeAnnotationPrinter::print_(type_identity<Counter>, [[maybe_unused]] const Counter& counter) {
+    // TODO
```
Comment (b-scholz) [inline src/ast/analysis/Type.cpp:681]: The Counter is just an intrinsic functor with no arguments and a number as a result. Perhaps we add this.

**Review by b-scholz — CHANGES_REQUESTED** [review]
Great work!

File: src/ast/analysis/Type.cpp (hunk 1 above)
Comment (hide-kawabata) [reply src/ast/analysis/Type.cpp]: The function has been added.

File: src/ast/analysis/Type.cpp (hunk 2 above)
Comment (hide-kawabata) [reply src/ast/analysis/Type.cpp]: Done, although in a bit superficial way.

File: src/ast/analysis/Type.cpp [hunk 3]
```diff
@@ -503,6 +510,256 @@ void TypeAnalysis::run(const TranslationUnit& translationUnit) {
...
+        }
+    }
+    os << ")";
+}
+
+void TypeAnnotationPrinter::print_(type_identity<Aggregator>, [[maybe_unused]] const Aggregator& agg) {
+    // TODO
+}
+
+// Note: branchOnArgument do the better job.
```
Comment (b-scholz) [inline src/ast/analysis/Type.cpp:732]: Please can you remove the Note and TODO above.

Comment (hide-kawabata) [reply src/ast/analysis/Type.cpp]: Instead of deleting those unfinished parts, I implemented functions for supporting aggregator. How is it?

**Review by b-scholz — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**